File: src/main/java/liqp/filters/Rstrip.java
package liqp.filters;

import liqp.nodes.RenderContext;

public class Rstrip extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        if (!super.isString(value)) {
            return value;
        }

        return super.asString(value).replaceAll("\\s+$", "");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Rstrip.kt
package liqp.filter

import liqp.context.LContext

class Rstrip : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return when(value) {
      is String-> value.trimEnd()
      else-> value
    }
  }
}
